src/deluge/dsp/compressor/rms_feedback.h

```diff
@@ -34,72 +34,144 @@ class RMSFeedbackCompressor {
+	[[nodiscard]] inline constexpr q31_t getAttack() const { return attackKnobPos; }
+
+	/// Get the current attack time constant in MS for a 3db change
+	///
+	/// @todo This currently rounds (down) to the nearest millisecond, it would probably be nice to expose at least a
+	/// little more of the underlying precsision.
+	[[nodiscard]] inline constexpr int32_t getAttackMS() const { return attackMS; }
+
+	/// Set the attack time constant from a full-scale (0 to 2^31) number.
+	///
+	/// The parameter is exponentially mapped to a number between 0.5 and 70-ish ms
 	constexpr int32_t setAttack(q31_t attack) {
 		// this exp will be between 1 and 7ish, half the knob range is about 2.5
 		attackMS = 0.5 + (std::exp(2 * float(attack) / ONE_Q31f) - 1) * 10;
 		a_ = (-1000.0f / kSampleRate) / attackMS;
 		attackKnobPos = attack;
 		return attackMS;
 	};
-	inline q31_t getRelease() { return releaseKnobPos; }
-	inline int32_t getReleaseMS() { return releaseMS; }
+
+	/// Get the current release time constant in terms of the full knob range (0 to 2^31)
+	[[nodiscard]] inline constexpr q31_t getRelease() const { return releaseKnobPos; }
+	/// Get the current release time constant in MS for a 3db change
+	///
+	/// @todo This currently rounds (down) to the nearest millisecond, it would probably be nice to expose at least a
+	/// little more of the underlying precsision.
+	[[nodiscard]] inline constexpr int32_t getReleaseMS() const { return releaseMS; }
+
+	/// Set the release time constant from a full-scale (0 to 2^31) number.
+	///
+	/// The parameter is exponentially mapped to a number between 50 and 400-ish ms
 	constexpr int32_t setRelease(q31_t release) {
 		// this exp will be between 1 and 7ish, half the knob range is about 2.5
 		releaseMS = 50 + (std::exp(2 * float(release) / ONE_Q31f) - 1) * 50;
 		r_ = (-1000.0f / kSampleRate) / releaseMS;
 		releaseKnobPos = release;
 		return releaseMS;
 	};
-	q31_t getThreshold() { return thresholdKnobPos; }
+
+	/// Get the current theshold as a full-scale (0 to 2^31) number.
+	[[nodiscard]] inline constexpr q31_t getThreshold() const { return thresholdKnobPos; }
+	/// Set the threshold based on a full-scale (0 to 2^31) number.
+	///
+	/// 0 corresponds to a threshold of 0.2, 2^31 corresponds to a theshold of 1.
 	constexpr void setThreshold(q31_t t) {
 		thresholdKnobPos = t;
 		threshold = 1 - 0.8f * (float(thresholdKnobPos) / ONE_Q31f);
 	}
-	q31_t getRatio() { return ratioKnobPos; }
-	constexpr int32_t getRatioForDisplay() { return ratio; }
+
+	/// Get the current ratio as a full-scale (0 to 2^31) number
+	[[nodiscard]] inline constexpr q31_t getRatio() const { return ratioKnobPos; }
+
+	/// Get the current ratio as an integer
+	///
+	/// @todo Thsi currently rounds (down) to the nearest integer, it would be nice to expose at least a little more of
+	/// the underlying precision.
+	[[nodiscard]] inline constexpr int32_t getRatioForDisplay() const { return ratio; }
```

Comment:
This should probably return a float so that we can display 2.x

Reply:
i think this should be done in a separate pr/as future work. the lack of precision in the ratio display has already confused people though, so we definitely should fix it.